[InfiniteRecursion] TypeDef.java

[ERROR] ../sundrio/model/base/src/main/java/io/sundr/model/TypeDef.java:[275,4] error: [InfiniteRecursion] This method always recurses, and will cause a StackOverflowError
[ERROR]     (see https://errorprone.info/bugpattern/InfiniteRecursion)

--- a/model/base/src/main/java/io/sundr/model/TypeDef.java
+++ b/model/base/src/main/java/io/sundr/model/TypeDef.java
@@ -272,7 +272,7 @@ public class TypeDef extends ModifierSupport implements Renderable, Nameable, An
    * @return a {@link ClassRef} for the currnet type, with the specified type arguments
    */
   public ClassRef toReference(Collection<TypeRef> arguments) {
-    return toReference(arguments);
+    return toReference(new ArrayList<>(arguments));
   }
 
   /**

